pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/DefaultMetadataStoreProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.metadata.impl;
+
+import org.apache.pulsar.metadata.api.MetadataStore;
+import org.apache.pulsar.metadata.api.MetadataStoreConfig;
+import org.apache.pulsar.metadata.api.MetadataStoreException;
+import org.apache.pulsar.metadata.api.MetadataStoreProvider;
+
+public class DefaultMetadataStoreProvider implements MetadataStoreProvider {
+
+    public interface MetadataStoreConstructor {
```

Comment:
I think this creates an extra hop of confusion - it took me a while to figure out what this is doing here.
When you see `MetadataStoreProvider`, you almost automatically expect to see the different implementations per store.
The fact that we indirectly place them in another interface named `MetadataStoreConstructor` is confusing.

I suggest having an implementation per store of `MetadataStoreProvider`. It's very short anyway, two methods, each having one line. 

IMO it will lower the confusion and be aligned with the external implementations, which will have one per store.